src/tbp/monty/frameworks/models/motor_policies.py

```diff
@@ -1105,129 +1244,12 @@ def pre_episode(self):
-        # be less efficient, but may be important for smaller/distant objects that
-        # otherwise get missed
-
-        if self.touch_search_amount >= 720:
-            # Perform a random upward or downward movement along the surface of a
-            # sphere, with its centre fixed 10 cm in front of the agent
-            logging.debug("Trying random search for object")
-            if self.rng.uniform() < 0.5:
-                orientation = "vertical"
-                logging.debug("Orienting vertically")
-            else:
-                orientation = "horizontal"
-                logging.debug("Orienting horizontally")
-
-            rotation_degrees = self.rng.uniform(-180, 180)
-            logging.debug(f"Random orientation amount is : {rotation_degrees}")
-
-        elif self.touch_search_amount >= 360 and self.touch_search_amount < 720:
-            logging.debug("Trying vertical search for object")
-            orientation = "vertical"
-
-        else:
-            logging.debug("Trying horizontal search for object")
-            orientation = "horizontal"
-
-        # Move tangentally to the point we're facing, resulting in the agent's
-        # orienation about the central point changing as specified by rotation_deg,
-        # but now where the agent is no longer on the edge of the circle
-        move_lat_amount = np.tan(np.radians(rotation_degrees)) * distance_from_center
-        # The lateral movement will be down relative to the agent's facing position
-        # in the case of orient vertical, and left in the case of orient horizontal
-
-        # The below calculates the exsecant, which provides the necessary
-        # movement to bring the agent back to the same radius around the central
-        # point
-        move_forward_amount = distance_from_center * (
-            (1 - np.cos(np.radians(rotation_degrees)))
-            / np.cos(np.radians(rotation_degrees))
-        )
-
-        self.touch_search_amount += rotation_degrees  # Accumulate total rotations
-        # for touch-search
-
-        return (
-            OrientVertical(
-                agent_id=self.agent_id,
-                rotation_degrees=rotation_degrees,
-                down_distance=move_lat_amount,
-                forward_distance=move_forward_amount,
-            )
-            if orientation == "vertical"
-            else OrientHorizontal(
-                agent_id=self.agent_id,
-                rotation_degrees=rotation_degrees,
-                left_distance=move_lat_amount,
-                forward_distance=move_forward_amount,
-            )
-        )
-
     ###
     # Methods that define behavior of __call__
     ###
     def dynamic_call(
-        self, state: Optional[MotorSystemState] = None
-    ) -> OrientHorizontal | OrientVertical | MoveTangentially | MoveForward | None:
+        self, state: MotorSystemState | None = None
+    ) -> MoveForward | OrientHorizontal | OrientVertical | MoveTangentially | None:
```

Comment:
just out of curiosity, what's the reason that these are reordered? Do we follow a specific convention here?

Reply:
Also, now that we raise ObjectNotVisible, is None still a possible output?

Reply:
It looks like it's possible for `self.get_next_action` to return `None`, so that's probably why it's still here. That does seem like a rare case though, looking at its definition.

Reply:
The SurfacePolicy action cycle is MoveForward->OrientHorizontal->OrientVertical->MoveTangentially, so I did it to match these. I think alphabetical order may make the most sense, though.

@jeremyshoemaker, you're right that it can still return `None` 😕. I'm gonna remove that and see what happens, as I don't think it should be doing that anymore.